File: algorithm/src/main/java/com/kkd/study/algorithm/leetcode/easy/_88.java
package com.kkd.study.algorithm.leetcode.easy;

/**
 * https://leetcode.com/problems/merge-sorted-array/
 */
public class _88 {
	public void merge(int[] nums1, int m, int[] nums2, int n) {
		int[] temp = new int[m + n];

		int i=0;
		int j=0;
		int k=0;

		while (i<m && j<n) {
			if (nums1[i] < nums2[j]) {
				temp[k++] = nums1[i++];
			} else {
				temp[k++] = nums2[j++];
			}
		}

		while (i<m) {
			temp[k++] = nums1[i++];
		}

		while (j<n) {
			temp[k++] = nums2[j++];
		}

		for (int idx=0 ; idx<k ; idx++) {
			nums1[idx] = temp[idx];
		}
	}
}

File: problem-solving/leetcode/88.go
package leetcode

func merge(nums1 []int, m int, nums2 []int, n int) {
	idx := m + n - 1
	for n > 0 {
		if m == 0 {
			nums1[idx] = nums2[n-1]
			idx--
			n--
			continue
		}

		if nums1[m-1] > nums2[n-1] {
			nums1[idx] = nums1[m-1]
			idx--
			m--
		} else {
			nums1[idx] = nums2[n-1]
			idx--
			n--
		}
	}
}
